katharsis-core/src/main/java/io/katharsis/errorhandling/mapper/KatharsisExceptionMapper.java

```diff
@@ -2,14 +2,25 @@
 
 import io.katharsis.errorhandling.ErrorResponse;
 import io.katharsis.errorhandling.exception.KatharsisMappableException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Exception mapper for a generic exception which can be thrown in request processing.
  */
 public final class KatharsisExceptionMapper implements JsonApiExceptionMapper<KatharsisMappableException> {
 
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
     @Override
     public ErrorResponse toErrorResponse(KatharsisMappableException exception) {
+        // log 5xx status as error and anything else as warn
+        if (exception.getHttpStatus() >= 500 || exception.getHttpStatus() < 600) {
+            logger.error("failed to process request", exception);
+        } else {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) This branch's code block is the same as the block for the branch on line 18. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1871)